aws-android-sdk-cognitoidentityprovider/src/main/java/com/amazonaws/mobileconnectors/cognitoidentityprovider/CognitoUser.java

```diff
@@ -925,7 +925,7 @@ public Runnable respondToMfaChallenge(final String mfaCode,
      * @return Valid, cached tokens {@link CognitoUserSession}. {@code null}
      *         otherwise.
      */
-    protected CognitoUserSession getCachedSession() {
+    public CognitoUserSession getCachedSession() {
```

Comment:
@androidsudo @bryannevega - How are you planning to consume this?
```
public boolean isSignedIn() {
    try {
        getCachedSession();
        return true;
    } catch (AmazonClientException sessionAcessFailure) {
        return false;
    }
}
```
Something like that? Would it be more useful to you if this method were cooked into the `CognitoUser`, so you didn't have to introduce it to your consuming code base(s)?

Reply:
@jamesonwilliams hello! sorry that missed your question! From the October 2019 i'm using own fork of cognito sdk,
How i use it - mostly as described [here](https://github.com/aws-amplify/aws-sdk-android/pull/1087#issuecomment-542445186) because of async nature of `public void getSession(final AuthenticationHandler callback)` some use cases (for example when i need to have JWT in some background tasks immediately) are requires to have syncing mechanics that's leads to code complexity increasing